### PR TITLE
add cosmovisor guide for newtork upgrade

### PR DESCRIPTION
**Pull request by allthatjazzleo**

### CODE REVIEW

File: docs/getting-started/cosmovisor.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+- Start `cosmovisor`, e.g.:
+
+```bash
+$ ./cosmovisor start
+```
+
+Since we enable `DAEMON_ALLOW_DOWNLOAD_BINARIES=true`, cosmovisor will automatically download binary with `2.0.1` in `./cosmovisor/upgrades/v2.0.0/bin/chain-maind` and update `./cosmovisor/current` directory symlink to `upgrades/v2.0.0` instead when proposed upgrade log is found. cosmovisor will create `./cosmovisor/upgrades/v2.0.0/bin/chain-maind` for you.
+
+:::warning Important:
+If one doesn't want to enable `DAEMON_ALLOW_DOWNLOAD_BINARIES`, one should prepare `./cosmovisor/upgrades/v2.0.0/bin/chain-maind` mamually before upgrade time.
```
Comment (aw126) [inline docs/getting-started/cosmovisor.md:59]:
```suggestion
If one doesn't want to enable `DAEMON_ALLOW_DOWNLOAD_BINARIES`, one should prepare `./cosmovisor/upgrades/v2.0.0/bin/chain-maind` manually before upgrade time.
```

File: docs/getting-started/cosmovisor.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# Running "Canis Major" network upgrade with cosmovisor
+
+## Step 0 - Install cosmosviosr
+
+One can follow this [link](https://docs.cosmos.network/master/run-node/cosmovisor.html#installation) to install.
+
+Or, run below command
```
Comment (aw126) [inline docs/getting-started/cosmovisor.md:7]:
```suggestion
Alternatively, run the below commands instead
```

File: docs/getting-started/cosmovisor.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+│   │   └── bin
+│   │       └── chain-maind
+│   └── upgrades
+│       └── v2.0.0
+│           └── bin
+│               └── chain-maind
+└── data
+```
+
+For `.chain-maind/cosmovisor/genesis/bin/chain-maind`, it is the binary before the upgrade proceeds. In our case of the "Canis Major" upgrade, it should be chain-maind with version `1.2.1`. Kindly, prepare this binary before run cosmovisor.
```
Comment (aw126) [inline docs/getting-started/cosmovisor.md:46]:
```suggestion
For `.chain-maind/cosmovisor/genesis/bin/chain-maind`, it is the binary before the upgrade proceeds. In our case of the "Canis Major" upgrade, it should be chain-maind with version `1.2.1`. Kindly have this binary ready before running cosmovisor.

File: docs/getting-started/upgrade_guide.md [hunk 4]
```diff
@@ -5,10 +5,13 @@ The chain-maind `v2.0.1` - Canis Major upgrade is proposed to be scheduled on 1
...
+`ERR UPGRADE "v2.0.0" NEEDED at time: 2021-05-25T23:59:00Z: {"binaries":{"darwin/amd64":"https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Darwin_x86_64.tar.gz?checksum=sha256:a4d51bf98350c7ecbb5e6bab192c9cac2f4059754e5507d2a1970a8a5488c74a","darwin/arm64":"https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Darwin_arm64.tar.gz?checksum=sha256:eb6274724eab1957a7b708a5a0887de9f213d053841fb58041f697e13728a454","linux/amd64":"https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Linux_x86_64.tar.gz?checksum=sha256:5e9e9f703cb85c72573086e384e187e752463b2ed0ccd612094a1f29a13f0158","linux/arm64":"https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Linux_arm64.tar.gz?checksum=sha256:345c7eacfc768df355c3b3ecadc9cc39e3c1656c9c0c4d465b938834fa66ff03","windows/amd64":"https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Windows_x86_64.zip?checksum=sha256:07b06382397df49774788ec06a5c9d0bf7e28ddb7b1dd91abdb35174bce49a31"}}`
```
Comment (allthatjazzleo) [inline docs/getting-started/upgrade_guide.md:14]:
```suggestion
`ERR UPGRADE "v2.0.0" NEEDED at time: 2021-06-01T23:59:00Z: {"binaries":{"darwin/amd64":"https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Darwin_x86_64.tar.gz?checksum=sha256:a4d51bf98350c7ecbb5e6bab192c9cac2f4059754e5507d2a1970a8a5488c74a","darwin/arm64":"https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Darwin_arm64.tar.gz?checksum=sha256:eb6274724eab1957a7b708a5a0887de9f213d053841fb58041f697e13728a454","linux/amd64":"https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Linux_x86_64.tar.gz?checksum=sha256:5e9e9f703cb85c72573086e384e187e752463b2ed0ccd612094a1f29a13f0158","linux/arm64":"https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Linux_arm64.tar.gz?checksum=sha256:345c7eacfc768df355c3b3ecadc9cc39e3c1656c9c0c4d465b938834fa66ff03","windows/amd64":"https://github.com/crypto-org-chain/chain-main/releases/download/v2.0.1/chain-main_2.0.1_Windows_x86_64.zip?checksum=sha256:07b06382397df49774788ec06a5c9d0bf7e28ddb7b1dd91abdb35174bce49a31"}}`
```

**Review by allthatjazzleo — COMMENTED** [review]
fix upgrade time

**Review by tomtau — COMMENTED** [review]
looks all right, but perhaps there should be a warning that this is better to be used on full nodes rather than validators -- e.g. currently there's a potential race condition: https://github.com/cosmos/cosmos-sdk/pull/9384

**Comment by allthatjazzleo** [comment]
> looks all right, but perhaps there should be a warning that this is better to be used on full nodes rather than validators -- e.g. currently there's a potential race condition: [cosmos/cosmos-sdk#9384](https://github.com/cosmos/cosmos-sdk/pull/9384)

added warning

**Review by lezzokafka — APPROVED** [review]
Thanks @allthatjazzleo !

**Pull request merged**